manual/gce-sru-20.2.45.txt

```diff
@@ -0,0 +1,1857 @@
++ local value=
++ for var in $@
++ eval 'value=$PROPOSED_SCRIPT > /dev/null'
+++ value=setup_proposed.sh
++ '[' -z setup_proposed.sh -o setup_proposed.sh = UNSET ']'
++ for var in $@
++ eval 'value=$SAMPLE_CLOUDCONFIG > /dev/null'
+++ value=sethostname.yaml
++ '[' -z sethostname.yaml -o sethostname.yaml = UNSET ']'
++ for var in $@
++ eval 'value=$SSHOPTS > /dev/null'
+++ value=-o
++ '[' -z -o -o -o = UNSET ']'
++ for var in $@
++ eval 'value=$GCE_ZONE > /dev/null'
+++ value=us-central1-a
++ '[' -z us-central1-a -o us-central1-a = UNSET ']'
++ for var in $@
++ eval 'value=$NETCFG_FILE > /dev/null'
+++ value=/etc/netplan/50-cloud-init.yaml
++ '[' -z /etc/netplan/50-cloud-init.yaml -o /etc/netplan/50-cloud-init.yaml = UNSET ']'
++ create_samplecloudconfig SAMPLE_CLOUDCONFIG
++ filename=SAMPLE_CLOUDCONFIG
++ cat
++ create_setup_proposed_script
++ '[' 0 -eq 0 ']'
++ cat
++ echo '### BEGIN focal'
+### BEGIN focal
++ case $UBUNTU_RELEASE in
++ family=ubuntu-2004-lts
++ gcloud compute instances create focal-sru-test --image-family ubuntu-2004-lts --image-project ubuntu-os-cloud-devel --metadata-from-file user-data=sethostname.yaml --zone=us-central1-a
+Created [https://www.googleapis.com/compute/v1/projects/ubuntu-server-277015/zones/us-central1-a/instances/focal-sru-test].
+NAME            ZONE           MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP     STATUS
+focal-sru-test  us-central1-a  n1-standard-1               10.128.0.14  35.223.168.228  RUNNING
+++ gcloud compute instances list
+++ grep focal-sru
+++ awk '{printf "%s", $5}'
++ VM_IP=35.223.168.228
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- cloud-init status --wait --long
+ssh: connect to host 35.223.168.228 port 22: Connection timed out
++ sleep 5
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- cloud-init status --wait --long
+
+status: done
+time: Tue, 23 Jun 2020 20:08:18 +0000
+detail:
+DataSourceGCE
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- dpkg-query --show cloud-init
+cloud-init	20.1-10-g71af48df-0ubuntu5
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- sudo cat /run/cloud-init/result.json
+{
+ "v1": {
+  "datasource": "DataSourceGCE",
+  "errors": []
+ }
+}
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- '!' grep Trace /var/log/cloud-init.log
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- systemd-analyze
+Startup finished in 1.265s (kernel) + 1min 27.081s (userspace) = 1min 28.347s
+graphical.target reached after 1min 24.019s in userspace
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@35.223.168.228 -- systemd-analyze blame
+1min 9.340s snapd.seeded.service
```

Comment:
I know this is not related to the work we are doing, but @blackboxsw is it expected for snapd.seeded to take that long in GCE ?

Reply:
I've seen it do that randomly before...actually during my "homework" for this position :smiley:

Reply:
yes it's definitely a pain point on most images, gce gets hit pretty hard by this.
previous SRU costs:
manual/gce-sru-19.3.41.txt:    1min 13.849s snapd.seeded.service
manual/gce-sru-19.3.41.txt:    1min 10.277s snapd.seeded.service